app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt

```diff
@@ -199,160 +268,202 @@ fun FilterScreen(
-              onStringChanged = { tagsViewModel.setQuery(it, filteringTags.value) },
-              textColor = MaterialTheme.colorScheme.onSurface,
-              onDropDownItemClicked = { tag -> tagsViewModel.addFilter(tag) },
-              height = tagSelectorHeight,
-              width = tagSelectorWidth,
-              textSize = (tagSelectorHeight.value * tagSelectorTextSizeFactor).sp)
-          Box(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
-            Button(
-                onClick = {
-                  filterViewModel.setFilteredTags(tagsViewModel.filteringTags.value)
-                  tagsViewModel.applyFilters()
-                  tagsViewModel.leaveUI()
-
-                  if (manSelected) currentSelectedGenders.add(Gender.MEN)
-                  if (womanSelected) currentSelectedGenders.add(Gender.WOMEN)
-                  if (otherSelected) currentSelectedGenders.add(Gender.OTHER)
-                  filterViewModel.setGenders(currentSelectedGenders)
-
-                  when {
-                    ageFrom != null && ageTo != null && !ageRangeError -> {
-                      filterViewModel.setAgeRange(ageFrom!!..ageTo!!)
-                    }
-                    ageFrom == null && ageTo == null -> {
-                      filterViewModel.setAgeRange(null)
-                    }
-                    ageFrom != null && ageTo == null -> {
-                      filterViewModel.setAgeRange(ageFrom!!..Int.MAX_VALUE)
-                    }
-                    ageFrom == null && ageTo != null -> {
-                      filterViewModel.setAgeRange(0..ageTo!!)
-                    }
-                    else -> {
-                      filterViewModel.setAgeRange(null)
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.Center,
+                        verticalAlignment = Alignment.CenterVertically,
+                    ) {
+                      GenderButton(
+                          selected = manSelected,
+                          onClick = {
+                            manSelected = !manSelected
+                            isModified = checkModified()
+                          },
+                          label = "Men")
+                      GenderButton(
+                          selected = womanSelected,
+                          onClick = {
+                            womanSelected = !womanSelected
+                            isModified = checkModified()
+                          },
+                          label = "Women")
+                      GenderButton(
+                          selected = otherSelected,
+                          onClick = {
+                            otherSelected = !otherSelected
+                            isModified = checkModified()
+                          },
+                          label = "Other")
                     }
+
```

Comment:
It maybe makes things simpler to create a list of labels ("Men", "Women", "Other") and generate the button for each label to have a cleaner implementation letting us having some modularity in the code.